### ISSUE TITLE
[SPIKE] Change "Attendee" field API name to "Contact" on Session Registration Object

### ISSUE THREAD

**Issue by sgoEma** [sgoEma/EventManagement#47]
Analyze what needs to be done to change the field API name (or create a new field instead of the Attendee)

**Comment by sgoEma** [comment]
SP: 3

**Comment by t-gul** [comment]
@sgoEma @sercanEmakina @secoEmakina 
Issue: Change "Attendee" field API name to "Contact" on Session Registration Object
Where it is Used :
Reference Type | Reference Label

FlexiPage | Session_Registration_Record_Page
Flow | After Insert Event Registration
Flow | Before Delete Event Regs
Flow | Before Save Session Registration Validations
Layout | Session Registration Layout
Layout | Contact (Attendee) Layout
Layout | Speaker Layout
Layout | Contact Layout

Problem:
It is not allowed for Users to change the API name of the Objects/Fields, if it is referenced in any of the metadata. The changing of API Name without removing references can result in errors being thrown as the operation will be unsupported.

Solution 1

Removal of the usage of Field/Object in the metadata, then the components will be needed to be changed in order to refer to the new API Name of the Field.

Solution 2

We can use tools like Salesforce Extension for VS Code which enables you to search for the references in the metadata after it has been retrieved from the Org. After doing the same, you can change the references to the old API Name of the Fields.

**Comment by sercanEmakina** [comment]
<img width="382" alt="Screen Shot 2022-04-14 at 10 12 53" src="https://user-images.githubusercontent.com/100766670/163333445-34af7837-6b9e-403a-ab74-7d683ddabdf5.png">
@t-gul  I noticed there are 2 "Before Save Session Registration Validations"  flows, I guess second one is a different flow , is that correct? or is it just a typo error ?

**Comment by t-gul** [comment]
@sercanEmakina It's just a type error. It is fixed

**Comment by sercanEmakina** [comment]
<img width="362" alt="Screen Shot 2022-04-14 at 10 40 49" src="https://user-images.githubusercontent.com/100766670/163338478-1fae20a5-dbc6-4853-a80a-0541990fb6ce.png">

@t-gul do we have a flow named "Prevent Duplicate Session Regs" ? is that flow being actively used? if so we need to check what kind of processes are executed in the flow, we might need to move the logic inside that flow to Before Save Session Registration flow or a new flow that runs in an appropriate trigger context.

Let's discuss it in the daily meeting.

**Comment by t-gul** [comment]
@sercanEmakina 
"Prevent Duplicate Session Regs" flow  is not actively used
The flow is  deleted from related org and related comment is updated.

**Comment by sgoEma** [comment]
@t-gul Thanks for the analysis. Are you sure that we must change the references for this change?

**Comment by t-gul** [comment]
@sgoEma Yes I'm sure. If we don't change them we can't see them at layout or flexipage, and  related flows can result in errors.

**Comment by sgoEma** [comment]
@t-gul Can you please try to change the name of the field directly in a test environment and share the results?

**Comment by t-gul** [comment]
@sgoEma When I changed the API name of field, It's automatically changed in flows and it doesn't result in errors

We must not change the references  for this change according to test which I did.

But in documentation said that "The changing of API Name without removing references can result in errors being thrown as the operation will be unsupported" 

https://help.salesforce.com/s/articleView?id=000314907&type=1

**Comment by sgoEma** [comment]
@t-gul I could not totally understand your plan about this. So the references should be removed first and then the field name should be changed is that correct? So will we implement the functions again after changing the name?

**Comment by t-gul** [comment]
@sgoEma Sorry for the confusion, 
As we mention on meeting, we can change "Attendee" field API name to "Contact" on Session Registration Object and it will be changed in related flows 
After the changing, tests case must be opened and tests must be done for related flows.

**Comment by sgoEma** [comment]
@t-gul thanks! I'm gonna move the ticket under "Ready for Deployment".
